internal/backend/local/backend.go

```diff
@@ -350,7 +350,7 @@ func (b *Local) opWait(
 
 		// try to force a PersistState just in case the process is terminated
 		// before we can complete.
-		if err := opStateMgr.PersistState(nil); err != nil {
+		if err := opStateMgr.PersistState(cancelCtx, nil); err != nil {
```

Comment:
This is one that I'm not sure about. Perhaps we should actually use a totally separate context here?

Reply:
We should. Canceling that operation was not possible before. It looks like the last line of defense against losing state changes. Canceling it midway could have bad consequences, right?

Reply:
I would either go with a separate context or maybe `stopCtx`? Though separate sounds like a better choice

Reply:
I created a separate context and documented this choice.